addons/mrp/tests/common.py

```diff
@@ -53,9 +52,40 @@ def generate_mo(cls, tracking_final='none', tracking_base_1='none', tracking_bas
         mo.action_confirm()
         return mo, bom_1, product_to_build, product_to_use_1, product_to_use_2
 
+    @classmethod
+    def setup_by_admin(cls):
+        super().setup_by_admin()
+        # User Data: mrp user and mrp manager
+        cls.user_mrp_user = cls.env['res.users'].sudo().create({
+            'name': 'Hilda Ferachwal',
+            'login': 'hilda',
+            'email': 'h.h@example.com',
+            'notification_type': 'inbox',
+            'groups_id': [
+                cls.env.ref('mrp.group_mrp_user').id,
+                # cls.env.ref('mrp.group_stock_user').id,
+                cls.env.ref('mrp.group_mrp_byproducts').id,
+                cls.env.ref('uom.group_uom').id,
+            ],
+            'company_id': cls.env.company.id,
+        })
+        cls.user_mrp_manager = cls.env['res.users'].sudo().create({
+            'name': 'Gary Youngwomen',
+            'login': 'gary',
+            'email': 'g.g@example.com',
+            'notification_type': 'inbox',
+            'groups_id': [
+                cls.env.ref('mrp.group_mrp_manager').id,
+                # cls.env.ref('mrp.group_stock_user').id,
+                cls.env.ref('mrp.group_mrp_byproducts').id,
+                cls.env.ref('uom.group_uom').id,
+            ],
+            'company_id': cls.env.company.id,
+        })
```

Comment:
Feedback about `ref` noted, I'll change the code.

It looks like `_create_new_internal_user` only allows one user per class, and we usually need 2 ("user" and "manager").

Regarding `mail_test_new_user`:
* this method requires admin permissions (we are creating a new user, after all)
* but there is no way to pass `sudo()` to this method at the moment
* so, the only way to call it is in `setup_independent_user`, before we configure the "stock" user
* but that doesn't allow the new user to be assigned to the "stock" company, because the company is created in the next step (`setup_independent_company`)
* so, I would have to:
  * create new users
  * create and switch to the "stock" user
  * create "stock" company
  * assign new users to "stock" company (in a different method than user creation)

It was either this, or modifying `mail_test_new_user` to use `.sudo()`, or creating the new users from scratch (which is the option I've chosen).

In general, using the existing API here created a lot of dependencies like this one. Perhaps I'm missing an obvious solution, in which case your help is appreciated ;)

Reply:
Ok, then the `mail_test_new_user` issue would be handled if we drop the use of `setup_independent_user`.

> It looks like `_create_new_internal_user` only allows one user per class, and we usually need 2 ("user" and "manager").

Why did you conclude that ? All those utils can be given any kwargs to override default values.  The default values are mostly there to help simplify the usage everywhere.